### PR TITLE
Fix/m2 1577 fix required password on every attempt to veiw respondent data

### PR DESCRIPTION
**Pull request by alex-i-e**

### CODE REVIEW

File: src/modules/Dashboard/features/Respondents/Respondents.tsx [hunk 1]
```diff
@@ -81,6 +89,15 @@ export const Respondents = () => {
       setDataExportPopupVisible(true);
     },
     viewDataAction: (index: number) => {
+      if (hasPassword && appletId) {
+        const respondentId = respondentsData?.result[index]?.id;
+        if (!respondentId) return;
+
+        navigate(generatePath(page.appletRespondentDataSummary, { appletId, respondentId }));
+
+        return;
```
Comment (kdanilenka32) [inline src/modules/Dashboard/features/Respondents/Respondents.tsx:98]: Why need a lot returns
 respondentId && navigate(generatePath(page.appletRespondentDataSummary, { appletId, respondentId }));

File: src/shared/hooks/useLogout.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ export const useLogout = () => {
 
   //TODO: rewrite to reset the global state data besides the data needed in lock form
   return () => {
-    storage.removeItem('workspace');
+    storage.clear();
```
Comment (kdanilenka32) [inline src/shared/hooks/useLogout.ts:14]: Are y sure to clear all ? What about lang?

File: src/shared/hooks/usePasswordFromStorage.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { useCheckIfNewApplet } from 'shared/hooks/useCheckIfNewApplet';
+import { auth } from 'modules/Auth';
+import { storage } from 'shared/utils';
+
+const getPasswordKey = (ownerId: string, appletId: string) => `pwd/${ownerId}/${appletId}`;
```
Comment (kdanilenka32) [inline src/shared/hooks/usePasswordFromStorage.ts:5]: pls move to utils

Comment (alex-i-e) [reply src/shared/hooks/usePasswordFromStorage.ts:5]: why ? I use it only in current hook

**Review by kdanilenka32 — APPROVED** [review]

**Review by IhorNesterenko99 — APPROVED** [review]

**Pull request merged**